### PR TITLE
Upgrade CLI to 12.0.0

### PR DESCRIPTION
**Pull request by szymonrybczak**
## Summary:

Upgrade to stable CLI release 🎉

## Changelog:

[General] [Changed] - Upgrade CLI to 12.0.0

## Test Plan:

CI

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -2366,98 +2366,118 @@
...
     joi "^17.2.1"
 
+"@react-native-community/cli-debugger-ui@12.0.0":
+  version "12.0.0"
+  resolved "https://registry.yarnpkg.com/@react-native-community/cli-debugger-ui/-/cli-debugger-ui-12.0.0.tgz#949e48f1770414d5d3c08d14ddd5b5847b6034a4"
+  integrity sha512-gOid9bGi9dfGm+Ro89SFY9gZfrEk29MFn8wETgEGZ3K+/lelGzysfZmXyV0qk/N5nNurL3jOyhHRvLqU+XGOdQ==
+  dependencies:
+    serve-static "^1.13.1"
+
 "@react-native-community/cli-debugger-ui@12.0.0-alpha.17":
```
Comment (tido64) [inline yarn.lock:2397]: We should figure out why an alpha version is still here. Like wise for `@react-native-community/cli-server-api`.

Comment (szymonrybczak) [reply yarn.lock]: Because `community-cli-plugin` right now also depends on CLI packages, and I forgot to change it there 😅 Thanks for catching this!

**Comment by szymonrybczak** [comment]
> Can we discard https://github.com/facebook/react-native/pull/41249 then?

This PR is to `0.73-stable` branch.

**Comment by szymonrybczak** [comment]
Closing since it is already bumped https://github.com/facebook/react-native/commit/126db5eb657ca919becbd14f6915e56262744176 😅

**Pull request closed without merging**